src/components/Station.tsx

```diff
@@ -1,12 +1,15 @@
 import React from 'react';
 import { useParams } from 'react-router-dom';
 
-const Station: React.FC = () => {
-    const { id } = useParams();
+export interface StationProps {
+    id: string;
+    name: string;
+}
+
+const Station: React.FC<StationProps> = (attributes) => {
```

Comment:
I would use destructing here so you directly reference id and name
```suggestion
const Station: React.FC<StationProps> = ({ id, name }) => {
```